### ISSUE TITLE
synch_gt.py 'already synced' guard false-positives when rgb_raw.csv exists from another tool

### ISSUE THREAD

**Issue by alejandrofontan** [VSLAM-LAB/VSLAM-LAB#125]
## Symptom

`pixi run synch-gt eth table_3` skips with:

```
[synch_gt.py] Skipping eth:table_3 - already synced (found *_raw.csv, use --revert first)
```

but the sequence is **not** synced: `rgb.csv` has 1180 rows while `groundtruth.csv` has 3066, and only ~848 rgb frames even have a gt sample within 50 ms (the gt only covers part of the rgb timeline).

## Cause

`synch_pair()` treats the existence of *either* `rgb_raw.csv` *or* `groundtruth_raw.csv` as proof that syncing already ran (`Datasets/extra-files/synch_gt.py:105-106`). On this local copy of `ETH/table_3`, `rgb_raw.csv` was created by a *different* processing step (its content differs from `rgb.csv` — the current file has depth/mask columns), and `groundtruth_raw.csv` doesn't exist. The guard can't distinguish "synch_gt ran" from "some other tool left an rgb backup".

Consequence: training via `droid_slam/data_readers/vslamlab.py` (which pairs `rgb.csv`/`groundtruth.csv` positionally by row) fails with a CUDA device-side assert in `compute_distance_matrix_flow` — the exact trap documented in `Baselines/DROID-SLAM-DEV/CLAUDE.md`'s fine-tuning gotchas.

## Suggested fix

Make the guard positive-proof rather than presence-of-any-backup, e.g. skip only when **both** backups exist, or verify the actual synced invariant (`rgb.csv` and `groundtruth.csv` have equal row counts) and warn loudly when backups exist but the invariant fails. `--revert` is also unsafe in this state: it would restore `rgb_raw.csv` over changes made by the other tool.

Found while smoke-testing DROID-SLAM-DEV training locally on 2026-08-17.
